### PR TITLE
feat: suggest wrong region in login error message

### PR DESCRIPTION
**Pull request by kappa90**
## Problem
- Split from #11926 
- After the Cloud EU launch, users might mistakenly login in the wrong data region.

NOTE: DO NOT MERGE until Cloud EU has been released.

## Changes
- Improve the error message to suggest having selected the wrong data region.

## How did you test this code?
- Updated tests

### CODE REVIEW

File: posthog/api/authentication.py [hunk 1]
```diff
@@ -86,7 +86,10 @@ def create(self, validated_data: Dict[str, str]) -> Any:
         )
 
         if not user:
-            raise serializers.ValidationError("Invalid email or password.", code="invalid_credentials")
+            raise serializers.ValidationError(
+                "Invalid email or password. Make sure you have selected the right region.",
```
Comment (benjackwhite) [inline posthog/api/authentication.py:90]: Personally I don't think we should do this in the backend. I think this makes more sense that the frontend intercepts the standard error and displays an additional Alert indicating that we now have multiple regions, which one they are currently trying to login to and an easy link to take them to the proper place.

**Review by benjackwhite — CHANGES_REQUESTED** [review]

File: posthog/api/authentication.py (hunk 1 above)
Comment (kappa90) [reply posthog/api/authentication.py:90]: Mmm makes sense!

Comment (benjackwhite) [reply posthog/api/authentication.py:90]: Also we should include that same special banner in other places like "forgot password"

**Comment by kappa90** [comment]
Reinstated inside of #11926

**Pull request closed without merging**